digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporter.scala

```diff
@@ -157,21 +213,96 @@ class TrafficSignCsvImporter extends CsvDataImporterOperations {
+
+    val (floatingSigns, trafficSigns) = signs.partition(_.roadLink.isEmpty)
+
+    val (additionalFloatPanelInfo, floatingSignInfo) = floatingSigns.partition{ sign =>
+      TrafficSignType.applyOTHValue(sign.propertyData.find(p => p.publicId == typePublicId).get.values.head.asInstanceOf[TextPropertyValue].propertyValue.toString.toInt).group == AdditionalPanels}
+
+    val (additionalPanelInfo, trafficSignInfo) = trafficSigns.partition{ sign =>
+      TrafficSignType.applyOTHValue(sign.propertyData.find(p => p.publicId == typePublicId).get.values.head.asInstanceOf[TextPropertyValue].propertyValue.toString.toInt).group == AdditionalPanels}
+
+    val floatingAdditionalPanels = additionalFloatPanelInfo.map {panel => AdditionalPanelInfo(panel.mValue, panel.linkId, panel.propertyData, panel.validityDirection, Some(Point(panel.lon, panel.lat)))}.toSet
+
+    val additionalPanels = additionalPanelInfo.map {panel => AdditionalPanelInfo(panel.mValue, panel.linkId, panel.propertyData, panel.validityDirection, Some(Point(panel.lon, panel.lat)))}.toSet
+
+    val usedAdditionalPanels = trafficSignInfo.flatMap { sign =>
+      val mValue = GeometryUtils.calculateLinearReferenceFromPoint(Point(sign.lon, sign.lat), sign.roadLink.get.geometry)
+      val signBearing = Some(GeometryUtils.calculateBearing(sign.roadLink.get.geometry, Some(mValue), Some(sign.roadLink.get.length)))
+      val signType = sign.propertyData.find(p => p.publicId == typePublicId).get.values.headOption.get.asInstanceOf[TextPropertyValue].propertyValue.toString.toInt
+      val filteredAdditionalPanel = trafficSignService.getAdditionalPanels(sign.linkId, sign.mValue, sign.validityDirection, signType, sign.roadLink.get.geometry, additionalPanels, sign.nearbyLinks)
+
+      if (filteredAdditionalPanel.size <= 3) {
+        val propertyData = trafficSignService.additionalPanelProperties(filteredAdditionalPanel) ++ sign.propertyData
+        trafficSignService.createFromCoordinates(IncomingTrafficSign(sign.lon, sign.lat, sign.roadLink.get.linkId, propertyData, sign.validityDirection, signBearing), sign.roadLink.get, sign.nearbyLinks)
+        filteredAdditionalPanel
+      } else Seq()
+    }
+
+    val usedAdditionalFloatPanels = floatingSignInfo.map { sign =>
+      val signType = sign.propertyData.find(p => p.publicId == typePublicId).get.values.headOption.get.asInstanceOf[TextPropertyValue].propertyValue.toString.toInt
+      val filteredAdditionalPanel = trafficSignService.getAdditionalPanels(sign.linkId, sign.mValue, sign.validityDirection, signType, sign.nearbyLinks.head.geometry, additionalPanels, sign.nearbyLinks)
+
+      if (filteredAdditionalPanel.size <= 3) {
+        val propertyData = trafficSignService.additionalPanelProperties(filteredAdditionalPanel) ++ sign.propertyData
+        trafficSignService.createFloatingFromCoordinates(IncomingTrafficSign(sign.lon, sign.lat, sign.nearbyLinks.head.linkId, propertyData, sign.validityDirection, sign.bearing), sign.nearbyLinks.head)
```

Comment:
I think we could directly call createFloatingWithoutTransaction without creating a new sign and then copying new values